lib/rubygems/commands/signout_command.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require 'rubygems/command'
+require 'rubygems/commands/query_command'
+
+class Gem::Commands::SignoutCommand < Gem::Commands::QueryCommand
+
+  def initialize
+    super 'signout', 'Signout from all the current sessions.'
+
+    remove_option('--name-matches')
+  end
+
+  def description # :nodoc:
+    'This `signout` command is used to sign out from all the current sessions to be'\
+    'able to signin using different set of credentials.'
+  end
+
+  def usage # :nodoc:
+    program_name
+  end
+
+  def execute
+    if Gem.configuration.unset_api_key! then
+      $stdout.puts "You have successfully signed out out from 'RubyGems.org'."
+    else
+      $stdout.puts 'You are not currently signed in.'
+    end
+  rescue StandardError
+    $stderr.puts "File '#{Gem.configuration.credentials_path}' must have readonly permission."\
+                 " Please make sure its writeable."
+  end
```

Comment:
A few things here:

1. Having side effects in a condition (such as `if Gem.configuration.unset_api_key!` attempting to delete a file) can make reasoning about code more difficult.
2. You can use if/elsif/else instead of relying on exceptions.


Also, things inheriting from `Gem::Commands` (such as this) use `say` and `alert`, `alert_warning`, `alert_error` instead of `$stdout.puts` and `$stderr.puts`. 

The documentation for these methods is [available in lib/rubygems/user_interaction.rb](https://github.com/shivabhusal/rubygems/blob/a8fcba0260e10ec5508175dc5356c1ec75d0931e/lib/rubygems/user_interaction.rb#L349-L379).

---

So, taking those into consideration, maybe something like this?

```ruby
def execute
  credentials_path = Gem.configuration.credentials_path

  if !File.exist?(credentials_path) then
    alert_error 'You are not currently signed in.'
  elsif !File.writable?(credentials_path) then
    alert_error "File '#{Gem.configuration.credentials_path}' is read-only."\
                " Please make sure it is writable."
  else
    Gem.configuration.unset_api_key!
    say 'You have successfully signed out from all sessions.'
  end
end
```

(@segiddins: I'm not sure about my wording with "You have successfully signed out from all sessions." above, but my understanding is that that same config file can apply to other servers. Is this correct? And any thoughts on how to phrase this?)